### ISSUE TITLE
Minor correction regarding IIFE

### ISSUE THREAD

**Issue by jonathanp** [addyosmani/essential-js-design-patterns#175]
First of all I want to thank you for writing a great book! I'm reading it right now and found a **minor** semantic issue in the chapter _The Module Pattern_. It says...

> The pattern is quite similar to an immediately-invoked functional expression (IIFE - see the section on namespacing patterns for more on this) except that an object is returned rather than a function.

I don't think the module pattern is _similar_ to an IIFE; it _uses_ an IIFE. And in an IIFE a function isn't necessarily _returned_, it's _invoked_. Don't you agree? :)

I'm gonna continue to enjoy the book now and I hope you don't mind me submitting besserwisser issues like this! ;)

**Comment by addyosmani** [comment]
> I don't think the module pattern is similar to an IIFE; it uses an IIFE. And in an IIFE a function isn't necessarily returned, it's invoked. Don't you agree? :)

I agree @jonathanp. Master should now reflect this error in the way I originally described it :)

> Isn't this more of a problem with encapsulation (with private members) per se rather than with The Module Pattern?

Coming back to this pattern a few years later, you're pretty right. This is more of a fundamental problem around encapsulation in JS than the module pattern itself. I wonder if there's a way we can better word this section. 

The current book doesn't dive into any of the nuance around ES2015+ and more recent thinking about privacy in JS (WeakMaps, private/public class fields etc), although I think that might need to wait for a larger refresh of the book. Is there a good way you could see us rephrasing that paragraph? A PR would be welcome or we could iterate on wording here.
